js/displayMainPageChallenges.js

```diff
@@ -0,0 +1,43 @@
+import { fetchAllChallenges } from './apiService.js';
+import Challenge from './challengeClass.js';
+
+const challengesListElement = document.querySelector(".challenges__list");
+
+let challengesArray = [];
+const displayedChallenges = 3;
+
+const displayNumberOfChallenges = async () => {
+    const { success, data: challenges, error } = await fetchAllChallenges();
+
+    if (!success) {
+        console.error("Failed to fetch challenges:", error);
+        challengesListElement.innerHTML = `<li>Error loading challenges. Please try again later.</li>`;
+        return;
+    }
+
+    challengesArray = challenges;
+
+    const sortedChallenges = [...challengesArray].sort((a, b) => b.rating - a.rating);
+    const challengesToDisplay = sortedChallenges.slice(0, 3);
+
+
+    for (let i = 0; i < challengesToDisplay.length; i++) {
+        const challengeData = challengesToDisplay[i];
+
+        const challenge = new Challenge(
+            challengeData.id,
+            challengeData.title,
+            challengeData.description,
+            challengeData.type,
+            challengeData.minParticipants,
+            challengeData.maxParticipants,
+            challengeData.rating,
+            challengeData.image,
+            challengeData.labels
+        );
+
+        challengesListElement.appendChild(challenge.createChallengeCard(true));
```

Comment:
> dist file is as it should in git ignore, good ! The class combo with "this.id" is erased, as long as it does not affect the functionality- it is good-right? It is required that we display challenges with the 3 highest ratings together- great. Nice to see the bookings in the if statements- it looks professional and dynamic ! I think this is great, even though I don`t yet understand if the erased ID is a good or bad thing :-D

Thank you!
I'm not sure about the class names yet, but I added a function to the class that adds or removes a button. This is to avoid conflicts with the button in the future, as they are different!